content/zh-tw/devops.md

```diff
@@ -0,0 +1,21 @@
+---
+title: DevOps
+status: Completed
+category: 概念
+tags: ["方法論", "", ""]
+---
+
+## 什麼是 DevOps {#what-it-is}
+
+DevOps 代表著由團隊主導應用程式開發（development）到正式維運（operations）完整過程的一種方法，所以它才被稱作 DevOps。DevOps 不只是一套技術的運用，更是文化與流程的全面轉變。DevOps 要求工程團隊專注在更小的元件，而非一次處理整個功能，並且減少交接的次數，因為頻繁的交接正是導致錯誤的常見原因。
+
+## 解決的問題 {#problem-it-addresses}
+
+<!-- Need to change the links after /zh-tw/tightly-coupled-architectures and /zh-tw/monolithic_apps are completed -->
+傳統上，在具有[緊耦合](/tightly-coupled-architectures/)[單體式應用程式](/monolithic_apps/)的複雜組織裡，工作通常被分散在不同的團隊之間。這導致工作需要多次的交接與漫長的前置時間（lead time）。每當一個元件或更新準備就緒時，它就會被放到另一個團隊的工作佇列中。由於每個人都只處理專案的一小部分，而這種方式會導致主導權的缺乏。他們的目標將會是把工作交給下一個團隊，而不是為客戶提供正確的功能，這顯然是本末倒置。
+
+當程式碼終於部署到正式環境，早已經歷了許多的開發人員與等待佇列。假如此時程式碼無法運行，追查問題來源就會是一件非常困難的事情。DevOps 顛覆了上述這種開發方式。
+
+## 如何幫助 {#how-it-helps}
+
+讓一個團隊可以主導應用程式完整的生命週期，會帶來最少的交接次數、降低部署到正式環境的風險、擁有更好的程式碼品質（如果團隊也負起程式碼在正式環境運行的責任），並且因為團隊擁有更多的自主權與主導權而有更高的員工滿意度。
```

Comment:
```suggestion
讓一個團隊可以主導應用程式完整的生命週期，會帶來最少的交接次數、降低部署到正式環境的風險、擁有更好的程式碼品質，因為團隊也負起程式碼在正式環境運行的責任，故團隊擁有更多的自主權與主導權而有更高的員工滿意度。
```

Reply:
If break down the sentence as comments:

> (Having one team own the entire lifecycle of an application) (results in) 
[minimized handoffs], [reduce risk when deploying into production], 
[better code quality] (as teams are also responsible for how code performs in production and increased employee satisfaction due to more autonomy and ownership.)

The sentence is with wrong English grammar because it missed the last conjunction.

In correct English grammar, the sentence should be:

> (Having one team own the entire lifecycle of an application) (results in) 
[minimized handoffs], [reduce risk when deploying into production], 
[better code quality] (as teams are also responsible for how code performs in production //adj.// ) and [increased employee satisfaction]( due to more autonomy and ownership //adj.//).

This translation is also discussed with other professional translator and refer to it team (https://glossary.cncf.io/it/devops/).